File: frontend/ios/Offrii/Models/Category.swift
import Foundation

struct Category: Codable, Identifiable, Equatable {
    let id: UUID
    let name: String
    let icon: String?
    let isDefault: Bool
    let position: Int
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, name, icon, position
        case isDefault = "is_default"
        case createdAt = "created_at"
    }
}

File: backend/rest-api/src/models/category.rs
use chrono::{DateTime, Utc};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, FromRow)]
pub struct Category {
    pub id: Uuid,
    pub name: String,
    pub icon: Option<String>,
    pub is_default: bool,
    pub position: i32,
    pub created_at: DateTime<Utc>,
}
